### ISSUE TITLE
deezer download

### ISSUE THREAD

**Issue by Marcioghiraldelli** [ytdl-org/youtube-dl#31075]
Hello, I notice that deezer download works well, but it only downloads 30 seconds of audio. Is there an alternative to this?

**Comment by dirkf** [comment]
See https://github.com/yt-dlp/yt-dlp/issues/1591.

**Comment by Marcioghiraldelli** [comment]
@dirkf  as I understand it, it will not be possible more than 30 seconds, right?

**Comment by dirkf** [comment]
The original PR, mentioned in the link, is apparently invalid, in that it no longer matches what the site does. It does contain a Blowfish crypto module that might still be useful, here or elsewhere.

The discussion in the linked issue indicates that it might be possible to develop a PR that extracts the full tracks without DMCA infringement but it would have to be carefully assessed. Personally, I won't be doing this but others are welcome to try, whether for personal use or as a PR, subject to the considerations I mentioned.

**Comment by 89z** [comment]
I did an extractor for this before. It doesn't require a license server, so I don't think you can call it DRM.

**Comment by Marcioghiraldelli** [comment]
@89z  I liked your idea.
Do you have information on how to use your extractor?

**Comment by 89z** [comment]
please provide example URL

**Comment by Marcioghiraldelli** [comment]
@89z https://www.deezer.com/br/track/1754288087

**Comment by LucBerge** [comment]
Hello,
I was able to fix the deezer extractor, and uncypher the files. My PR was never merged and then the repo was taken down. I'll try to fix it in September.

See #22074

**Comment by 89z** [comment]
> I'll try to fix it in September.

I don't know if thats a good idea. I support a Deezer extractor, in fact I wrote my own. However its clear that Deezer will pursue some implementations. However I have also seen other valid implementations, that are pretty old and still standing. I guess we can see if a maintainer wants to give their opinion.

**Comment by LucBerge** [comment]
> Correction. It seems the DMCA in question was for YouTube-DL itself, and some forks:
> 
> https://github.com/github/dmca/blob/master/2020/10/2020-10-23-RIAA.md
> 
> since we all know that challenge ultimately failed, then it might be fine to add a Deezer extractor at this point, since it was RIAA and not Deezer that made the claim.

Indeed, it wasn't taken down because of me. I should have specify it...

**Comment by LucBerge** [comment]
Wait a minute, [my code was merged to yt-dlp from my fork of youtube-dl](https://github.com/yt-dlp/yt-dlp/commits/7a5c1cfe93924351387b44919b3c0b2f66c4b883/youtube_dlc/extractor/deezer.py?browsing_rename_history=true&new_path=yt_dlp/extractor/deezer.py&original_branch=86e5f3ed2e6e71eb81ea4c9e26288f16119ffd0c)

**Comment by dirkf** [comment]
As above, the question is whether PR #22074 still works.

Easiest is if you can get your repo unblocked (quote #27013), test the branch and commit any necessary changes to match the site there. Otherwise you can clone the PR code into a new branch and make a replacement PR from it (with whatever necessary changes).

**Comment by LucBerge** [comment]
@dirkf I could not get my repo unlocked. Check this [PR](https://github.com/yt-dlp/yt-dlp/pull/4935)
